src/zimcheck/checks.cpp

```diff
@@ -601,6 +601,8 @@ class TaskDispatcher
     explicit TaskDispatcher(ArticleChecker* ac, unsigned n)
         : currentCluster(-1)
     {
+        if (n > 8 )
+            throw std::runtime_error("Too many threads requested");
```

Comment:
Actually even this limit is quite high, given the initial experiments. zimcheck's run time doesn't scale well with more threads when the `--url_internal` check is enabled. But I think I better remove this limit rather than check it when parsing the command line.